src/common/components/userMatchModal/userMatchModal.component.js

```diff
@@ -166,6 +174,7 @@ export default angular
       modalTitle: '=',
       setLoading: '&',
       onStateChange: '&',
-      onSuccess: '&'
+      onSuccess: '&',
+      onFailure: '&'
```

Comment:
Oh. Remove this. It's probably overwriting your function. These bindings make things available on `this`. Since the template using using the `on-failure` binding, it's probably undefined.

Reply:
If I had not defined an `onFailure` function, what would it call? I see several places where `onSuccess` is being called, but is not defined in the controller.

Reply:
Bindings are element attributes that come from where the component is used. Somewhere there's a `<user-match-modal on-success="some angular expression">`. That expression would be run when `this.onSuccess()` is called.